src/AwsS3Adapter.php

```diff
@@ -438,7 +448,7 @@ public function listContents($dirname = '', $recursive = false)
      */
     protected function normalizeResponse(array $object, $path = null)
     {
-        $result = ['path' => $path ?: $this->removePathPrefix($object['Key'])];
+        $result = ['path' => $path ?: $this->removePathPrefix(isset($object['Key']) ? $object['Key'] : $object['Prefix'])];
```

Comment:
Sorry for the delay. I've created a gist showing how the response from S3 looks like:
https://gist.github.com/zaak/6b0788dfbc1391849dbb, should be more clear.
I have a set of tests in our app that use Flysystem quite intensively, so I assume this change shouldn't break anything.